lib/stripe/util.rb

```diff
@@ -337,44 +322,6 @@ def self.secure_compare(a, b)
     }.freeze
     private_constant :COLOR_CODES
 
-    # We use a pretty janky version of form encoding (Rack's) that supports
-    # more complex data structures like maps and arrays through the use of
-    # specialized syntax. To encode an array of maps like:
-    #
-    #     [{a: 1, b: 2}, {a: 3, b: 4}]
-    #
-    # We have to produce something that looks like this:
-    #
-    #     arr[][a]=1&arr[][b]=2&arr[][a]=3&arr[][b]=4
-    #
-    # The only way for the server to recognize that this is a two item array is
-    # that it notices the repetition of element "a", so it's key that these
-    # repeated elements are encoded first.
-    #
-    # This method is invoked for any arrays being encoded and checks that if
-    # the array contains all non-empty maps, that each of those maps must start
-    # with the same key so that their boundaries can be properly encoded.
-    def self.check_array_of_maps_start_keys!(arr)
```

Comment:
This method was designed to check that arrays being encoded were compatible with Rack syntax for arrays of maps — now that we're using integer-indexed arrays everywhere, we don't need to check this anymore.